### ISSUE TITLE
SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0 Error: -Xverbosegclog number of cycles must be greater than 0

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse/openj9#7969]
Failure link
------------

https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/214/tapResults/
ub16x64j95

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------

```
run_workload() - Command: /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/openjdkbinary/j2sdk-image/bin/java -Xcompressedrefs -Xmx1g -Xms1g -Xdump:none -Xgcpolicy:gencon -Xnocompactgc -Xgcthreads16 -Xgc:tenureBytesDeviationBoost=0 -Xgc:concurrentKickoffTenuringHeadroom=0 -Xgc:concurrentSlack=macrofrag -Xverbosegclog:/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/openjdk-tests/TKG/../TKG/test_output_15755241425800/SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0/config_10k_0.4.xml_2019-12-05T_03-25-20_verbose_concurrentSlackAuto.xml -cp .:SyntheticGCWorkload.jar net.adoptopenjdk.casa.workload_sessions.Main config/config_10k_0.4.xml --log_file /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/openjdk-tests/TKG/../TKG/test_output_15755241425800/SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0/config_10k_0.4.xml_2019-12-05T_03-25-20_stdout_concurrentSlackAuto.txt -s
JVMVERB004E Error: -Xverbosegclog number of cycles must be greater than 0
JVMJ9VM015W Initialization error for library j9vrb29(3): Failed to initialize
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
cat: /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/openjdk-tests/TKG/../TKG/test_output_15755241425800/SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0/config_10k_0.4.xml_2019-12-05T_03-25-20_stdout_concurrentSlackAuto.txt: No such file or directory
Workload Failure: test was not completed
FAILED

SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0_FAILED
```

Similarly SyntheticGCWorkload_concurrentSlackAuto_10M_J9_0

**Comment by pshipton** [comment]
@llxia @smlambert a TKG related issue? It didn't find the config file.

**Comment by smlambert** [comment]
Not sure that is the biggest problem in that output...

does this happen in healthy runs?

JVMJ9VM015W Initialization error for library j9vrb29(3): Failed to initialize
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.

**Comment by JasonFengJ9** [comment]
Last passing build [1] doesn't have such messages, `No such file or directory` or `j9vrb29(3): Failed to initialize`.

[1] https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/213/consoleFull

**Comment by pshipton** [comment]
@smlambert I expect the `JVMJ9VM015W Initialization error for library j9vrb29` is caused by `JVMVERB004E Error: -Xverbosegclog number of cycles must be greater than 0`, which in turn is caused by the problem with the `-Xverbosegclog:` parameter.

**Comment by llxia** [comment]
The error looks like the expected file was not generated.

The test passed in Grinder: https://ci.eclipse.org/openj9/view/Test/job/Grinder/564/
And it passed in yesterday nightly build: https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/215/

**Comment by pshipton** [comment]
Yup, we can consider it some weird hiccup. I'll close this until it occurs again.

**Comment by JasonFengJ9** [comment]
Another one today at https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/216/tapResults/

```
run_workload() - Command: /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/openjdkbinary/j2sdk-image/bin/java -Xcompressedrefs -Xmx1g -Xms1g -Xdump:none -Xgcpolicy:gencon -Xnocompactgc -Xgcthreads16 -Xgc:tenureBytesDeviationBoost=0 -Xgc:concurrentKickoffTenuringHeadroom=0 -Xgc:concurrentSlack=macrofrag -Xverbosegclog:/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/openjdk-tests/TKG/../TKG/test_output_15757981595427/SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0/config_10k_0.4.xml_2019-12-08T_07-33-21_verbose_concurrentSlackAuto.xml -cp .:SyntheticGCWorkload.jar net.adoptopenjdk.casa.workload_sessions.Main config/config_10k_0.4.xml --log_file /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/openjdk-tests/TKG/../TKG/test_output_15757981595427/SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0/config_10k_0.4.xml_2019-12-08T_07-33-21_stdout_concurrentSlackAuto.txt -s
JVMVERB004E Error: -Xverbosegclog number of cycles must be greater than 0
JVMJ9VM015W Initialization error for library j9vrb29(3): Failed to initialize
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
cat: /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/openjdk-tests/TKG/../TKG/test_output_15757981595427/SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0/config_10k_0.4.xml_2019-12-08T_07-33-21_stdout_concurrentSlackAuto.txt: No such file or directory
Workload Failure: test was not completed
FAILED

SyntheticGCWorkload_concurrentSlackAuto_10k_J9_0_FAILED
```

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/218
https://ci.eclipse.org/openj9/job/Test_openjdk13_j9_extended.functional_x86-64_linux_Nightly/111

https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/219

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/220

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/221

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/223

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/224

**Comment by llxia** [comment]
Grinder 10x and all passed: https://ci.eclipse.org/openj9/view/Test/job/Grinder/581/

**Comment by llxia** [comment]
- From the doc, the error [JVMVERB004E](https://www.ibm.com/support/knowledgecenter/SSYKE2_7.1.0/com.ibm.java.messages/diag/appendixes/messages/verb/messages_verb.html) is about
> Either you specified the -Xverbosegclog parameter with a value of 0 for the number of cycles, or the parameter could not be parsed.

Talked with @LinHu2016 , we suspect it was the path issue. We increased the log path and running a 20x Grinder:  https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder_Sandbox/36/

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/225

**Comment by LinHu2016** [comment]
looks like it is potential bug in port lib code, if the full path length is 255 byte, it might cause the parser failed.

I did run the test manually on one linux machine, get the same failure.

/bluebird/builds/bld_435392/sdk/xa6480/jre/bin/java -Xcompressedrefs -Xcompressedrefs -Xcompressedrefs -Xmx1g -Xms1g -Xdump:none -Xgcpolicy:gencon -Xnocompactgc -Xgcthreads16 -Xverbosegclog:/tmp/bld_435392/syntheticGCWorkload/625831633/log/config_10M_0.3.xml_2019-12-18T_12-14-21_verbose_original000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.xml -cp .:SyntheticGCWorkload.jar com.ibm.casa.workload_sessions.Main config/config_10M_0.3.xml --log_file log/config_10M_0.3.xml_2019-12-18T_12-14-21_stdout_original.txt -s
JVMVERB004E Error: -Xverbosegclog number of cycles must be greater than 0
JVMJ9VM015W Initialization error for library j9vrb29(3): Failed to initialize
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.

**Comment by llxia** [comment]
This issue is intermittent because the test output dir (i.e., test_output_15757981595427) is string concatenation of `time()` and `rand(10000)`, so the length may vary for every test run.

https://github.com/AdoptOpenJDK/TKG/blob/f3cd91b06f4ef534d701beb1f5b80480b2809a9b/scripts/getUniqueId.pl#L18-L20

The test will fail if the whole path length is 255 byte.

**Comment by pshipton** [comment]
@dmitripivkine fyi, assigned to gc.

**Comment by dmitripivkine** [comment]
@LinHu2016 Would you please fix?

**Comment by pshipton** [comment]
Resolved by #8152

**Comment by pshipton** [comment]
@dmitripivkine should we fix this in the 0.18 release?

**Comment by dmitripivkine** [comment]
> @dmitripivkine should we fix this in the 0.18 release?

@pshipton Well it is not a regression. It is up to you. Please let me know if you need help with this.

**Comment by pshipton** [comment]
It's causing a test failure in the release testing. Since there are no objections, I've cherry picked the fix for the release. https://github.com/eclipse/openj9/pull/8192
